osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/DeviceChannelsHelper.java

```diff
@@ -111,7 +115,7 @@ private String readIdentificationNumber(final List<GetResult> resultList, final
             throws ProtocolAdapterException {
 
         final Long identification = this.dlmsHelper.readLong(resultList.get(index), description);
-        return IdentificationNumber.fromIdentification(identification).getLast8Digits();
+        return identification.toString();
```

Comment:
DlmsHelper.readLong can return null:
```
    public Long readLong(final DataObject resultData, final String description) throws ProtocolAdapterException {
        final Number number = this.readNumber(resultData, description);
        if (number == null) {
            return null;
        }
        return number.longValue();
    }
```

So`identification.toString()` can cause an NPE. I think this should be fixed. For example by introducing Optional or throwing an exception.

Reply:
Changed readLong to readLonNotNull